packages/pigment-css-nextjs-plugin/src/index.ts

```diff
@@ -76,6 +70,12 @@ export function withPigment(nextConfig: NextConfig, pigmentConfig?: PigmentOptio
           if (what.startsWith('@emotion/styled') || what.startsWith('styled-components')) {
             return require.resolve('../third-party-styled');
           }
+          // Need to point to the react from node_modules during eval time.
+          // Otherwise, next makes it point to its own version of react that
+          // has a lot of RSC specific logic which is not actually needed.
+          if (what.startsWith('@babel') || what.startsWith('react') || what.startsWith('next')) {
```

Comment:
Is this strict enough? Would there be a risk matching on e.g. `react-hook-form`, `next-auth`,...

```suggestion
          if (what.startsWith('@babel/') || what.startsWith('react/') || what.startsWith('react-dom/') || what.startsWith('next/')) {
```